cli/cli
# Update IsEnterprise and IsTenancy for orthogonality using go-gh

Closes github/cli#577

## Changes

- Add test coverage to places where IsEnterprise incorrectly covers Tenancy
- Upgrade `go-gh` to v2.11.0
- Use `go-gh/auth` package for `IsEnterprise`, `IsTenancy`, and `NormalizeHostname`
- auth: Removed redundant `ghauth.IsTenancy(host)` check

This adds test coverage to ensure orthogonality between `IsEnterprise` and `IsTenancy` functionality, then imports the functions from `go-gh` to fix the tests. There's some additional cleanup beyond this as well. See the commit messages for more details.

## Testing
 
1. Pull down branch `gh pr checkout 9755`
2. Run all tests
3. Login and logout with a tenancy stamp to verify the flow is working: `gh auth login -h <tenancy-instance>`
4. Login and logout of a server stamp to verify the flow is working: `gh auth login -h <server-instance>`
